### PR TITLE
docs: Add more to Shader Validation docs

### PR DESCRIPTION
**Pull request by spencer-lunarg**
While refactoring shader validation I have been improving this, but figured it didn't make sense in any of the other PRs

### CODE REVIEW

File: docs/shader_validation.md [hunk 1]
```diff
@@ -15,6 +15,18 @@ All of these validations are passed off to `spirv-val` in [SPIRV-Tools](https://
 ## spirv-opt
 
 There are a few special places where `spirv-opt` is run to reduce recreating work already done in `SPIRV-Tools`.
+These can be found searching `RegisterPass` in the code
+
+Currently this is
```
Comment (arno-lunarg) [inline docs/shader_validation.md:20]:
```suggestion
Currently there are
```

File: docs/shader_validation.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ All of these validations are passed off to `spirv-val` in [SPIRV-Tools](https://
 ## spirv-opt
 
 There are a few special places where `spirv-opt` is run to reduce recreating work already done in `SPIRV-Tools`.
+These can be found searching `RegisterPass` in the code
```
Comment (arno-lunarg) [inline docs/shader_validation.md:18]:
```suggestion
These can be found searching for `RegisterPass` in the code
```

File: docs/shader_validation.md [hunk 3]
```diff
@@ -15,6 +15,18 @@ All of these validations are passed off to `spirv-val` in [SPIRV-Tools](https://
...
+
+Currently this is
+
+- Specialization constants
+  - This is used to inject the constants from the pipeline layout. Some checks require the runtime spec constant values
```
Comment (arno-lunarg) [inline docs/shader_validation.md:23]: Maybe just me, but what `This` refers to here? A run of `spirv-opt`
Also, reading 
>- Most likely will never be used (were deprecated early), but will catch if the app uses them
>- instruments the shaders
I do not understand those sentences, again maybe just me lol

File: docs/shader_validation.md [hunk 4]
```diff
@@ -31,7 +43,20 @@ The code is currently split up into the following main sections
...
     - This is a general util file that is [generated](generated_code.md) from the SPIR-V grammar
 
-### Design details
+## Types of Shader Validation
+
+All Shader Validation can be broken into 4 types of checks
+
+- Standalone with runtime properties
+  - Things like features and limits
+- Shader interfacing
```
Comment (arno-lunarg) [inline docs/shader_validation.md:52]:
```suggestion
- Shader interface
```
?

File: docs/shader_validation.md (hunk 2 above)
Comment (ncesario-lunarg) [inline docs/shader_validation.md:18]:
```suggestion
These can be found by searching for `RegisterPass` in the code
```

File: docs/shader_validation.md [hunk 5]
```diff
@@ -15,6 +15,18 @@ All of these validations are passed off to `spirv-val` in [SPIRV-Tools](https://
...
 
 There are a few special places where `spirv-opt` is run to reduce recreating work already done in `SPIRV-Tools`.
+These can be found searching `RegisterPass` in the code
+
+Currently this is
+
+- Specialization constants
+  - This is used to inject the constants from the pipeline layout. Some checks require the runtime spec constant values
+- Flatten OpGroupDecorations
+  - Most likely will never be used (were deprecated early), but will catch if the app uses them
```
Comment (ncesario-lunarg) [inline docs/shader_validation.md:25]: Maybe something along the lines of
```suggestion
  - Detects if group decorations were used; however, group decorations were [deprecated](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpGroupDecorate) early on in the development of the SPIR-v specification.
```
?

File: docs/shader_validation.md [hunk 6]
```diff
@@ -15,6 +15,18 @@ All of these validations are passed off to `spirv-val` in [SPIRV-Tools](https://
...
+- Debug Printf
+  - instruments the shaders
```
Comment (ncesario-lunarg) [inline docs/shader_validation.md:27]: Maybe reference docs/gpu_validation.md?

**Review by ncesario-lunarg — APPROVED** [review]
Some minor suggestions, but LGTM!

**Pull request merged**